src/components/structures/FilePanel.tsx

```diff
@@ -131,9 +141,7 @@ class FilePanel extends React.Component<IProps, IState> {
 
     public componentWillUnmount(): void {
         const client = MatrixClientPeg.get();
-        if (client === null) return;
-
-        if (!client.isRoomEncrypted(this.props.roomId)) return;
+        if (client === null || !this.state.isRoomEncrypted) return;
 
         if (EventIndexPeg.get() !== null) {
             client.removeListener(RoomEvent.Timeline, this.onRoomTimeline);
```

Comment:
Since `componentDidMount` is async, I think there's another potential for races here: the component mounts, then if it unmounts before `componentDidMount` even finishes determining whether the room is encrypted, it will set up event listeners that will never get removed.